src-tauri/src/utils.rs

```diff
@@ -0,0 +1,58 @@
+use std::{
+    fs::File,
+    io::{copy, Cursor},
+};
+use strum::Display;
+
+use crate::{error::AppError, ASSETS_DIR};
+
+/// Downloads an asset to the system-specific data directory.
+/// The file is saved into the `assets/tmp` folder.
+/// Since the `/tmp` folder is cleared on each app restart,
+/// the downloaded file needs to be further treated and moved out of the `/tmp` folder.
+///
+/// Restrictions:
+/// - max. file size: 2MB
+/// - supported file types: `.png`, `.svg`
+pub async fn download_asset(url: &str, logo_type: LogoType) -> Result<(), AppError> {
+    let url: reqwest::Url = url.parse().unwrap();
```

Comment:
Probably cleaner to change this to:
```
pub async fn download_asset(url: reqwest::Url, logo_type: LogoType) -> Result<(), AppError> {
```

or alternatively replace the `unwrap()` for an error message:
```
    let url: reqwest::Url = url
        .parse()
        .map_err(|_| AppError::DownloadAborted("Could not parse URL string."))?;
```